src/Aspire.Hosting.Azure.ServiceBus/ServiceBusHealthCheck.cs

```diff
@@ -38,9 +39,22 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
 
             var queueOrTopicName = _nameFactory();
 
-            if (!string.IsNullOrWhiteSpace(queueOrTopicName))
+            if (queueOrTopicName is null)
             {
-                var receiver = serviceBusClient.CreateReceiver(queueOrTopicName);
+                return HealthCheckResult.Healthy();
+            }
+
+            if (queueOrTopicName is AzureServiceBusQueueResource azureServiceBusQueueResource)
+            {
+                var receiver = serviceBusClient.CreateReceiver(azureServiceBusQueueResource.QueueName);
+                await receiver.PeekMessageAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+            if (queueOrTopicName is AzureServiceBusTopicResource azureServiceBusTopicResource
+                && azureServiceBusTopicResource.Subscriptions.Count != 0)
+            {
+                var receiver = serviceBusClient.CreateReceiver(azureServiceBusTopicResource.TopicName,
+                    azureServiceBusTopicResource.Subscriptions[0].SubscriptionName);
```

Comment:
@illay1994 The issue is that for topics we have to pass a subscription name?

Would you be able to copy the current unit tests that uses a queue to verify the health check then and change it to use a topic? Thanks